sns/parity_math.c

```diff
@@ -670,6 +858,168 @@ static void xor_recover(struct m0_parity_math *math,
         }
 }
 
+#ifndef __KERNEL__
+static void isal_recover(struct m0_parity_math *math,
+			 struct m0_buf *data,
+			 struct m0_buf *parity,
+			 struct m0_buf *fails,
+			 enum m0_parity_linsys_algo algo)
+{
+	uint8_t *fail;
+	uint32_t fail_count;
+	uint32_t unit_count = math->pmi_data_count + math->pmi_parity_count;
+	uint32_t block_size = data[0].b_nob;
+	uint8_t *temp_matrix = NULL;
+	uint8_t *invert_matrix = NULL;
+	uint8_t *decode_matrix = NULL;
+	uint8_t **data_in = NULL;
+	uint8_t **data_out = NULL;
+	uint32_t ui, i, j, r;
+	uint8_t *err_list = NULL;
+	uint8_t s;
+	int ret;
+
+	fail = (uint8_t*) fails->b_addr;
+	fail_count = fails_count(fail, unit_count);
+
+#if DEBUG
+	printf("\n fail_count: %d", fail_count);
+#endif
```

Comment:
No need to use printf();

M0_LOG(M0_ALWAYS, ...) is good enough.
Or
m0_console_printf() is also OK.

Reply:
Done